docs/api/specs/webConnectionProtocol.md

```diff
@@ -0,0 +1,382 @@
+
+To discover a Desktop Agent Preload interface, check for the presence of an `fdc3` object in the global scope (i.e. `window.fdc3`). If it exists return it immediately. If it does not exist then add a listener for the global `fdc3Ready` event with the the specified timeout, e.g.:
+
+```ts
+const discoverPreloadDA = async (timeout): Promise<DesktopAgent> => {
+  return new Promise((resolve, reject) => {
+    // if the global is already available resolve immediately
+    if (window.fdc3) {
+      resolve(window.fdc3);
+    } else {
+      //`fdc3Ready` event listener function
+      const listener = () => {
+        clearTimeout(timeout);
+        if (window.fdc3) {
+          clearTimeout(timeout);
+          resolve(window.fdc3);
+        } else {
+          reject('The `fdc3Ready` event fired, but `window.fdc3` Was not set!');
+        }
+      };
+      // Setup a timeout to return a rejected promise
+      const timeout = setTimeout(
+        () => {
+          //clear the event listener to ignore a late event
+          window.removeEventListener('fdc3Ready', listener);
+          if (window.fdc3){ 
+            resolve(window.fdc3);
+          } else {
+            reject('Desktop Agent Preload not found ');
+          }
+        }, 
+        timeout
+      );
+      // listen for the fdc3Ready event
+      window.addEventListener('fdc3Ready', listener, { once: true });
+    }
+  });
+};
+```
+
+To discover a Desktop Agent Proxy interface, locate all candidates for a `parent` window or frame by first checking the values of `window.opener` and `window.parent`. If `window.opener !== null` or `window.parent !== window` then they are candidates for a parent window or frame. As iframes can be nested we can also search for candidates that are parents of a parent frame, e.g.:
+
+```ts
+const discoverProxyCandidates = (): WindowProxy[] => {
+  const candidates = [];
+  //parent window
+  if (!!window.opener) { candidates.push(window.opener); }
+
+  //parent frames
+  let currentWin = window;
+  while (currentWin.parent !== currentWin){
+    candidates.push(currentWin.parent);
+    currentWin = currentWin.parent;
+  }
+
+  //parent window of top-level parent frame
+  if (window !== currentWin && !!currentWin.opener) { 
+    candidates.push(currentWin.opener); 
+  }
+  return candidates;
+}
+```
+
+Setup a timer for specified timeout, and then for each `candidate` found, attempt to establish communication with it as follows:
+
+  1. Add a listener (`candidate.addEventListener("message", (event) => {})`) to receive response messages from the `candidate`.
+  2. Send a [`"WCP1Hello"`](https://fdc3.finos.org/schemas/next/api/WCP1Hello.schema.json) message to it via `postMessage`.
+
+  ```ts
+  const hello = {
+      type: "WCP1Hello",
+      payload: {
+          identityUrl: identityUrl,
+          actualUrl: actualUrl,
+          fdc3Version: "2.2",
+          intentResolver: true
```

Comment:
Missing comma after   intentResolver: true

Reply:
fixed in https://github.com/finos/FDC3/commit/01f862f6cbfbeee77b31678fbbc9c6648940a45f